ts/client/src/accounts/bank.ts

```diff
@@ -481,6 +475,39 @@ export class Bank implements BankForHealth {
   getDepositRateUi(): number {
     return this.getDepositRate().toNumber() * 100;
   }
+
+  getNetBorrowLimitPerWindow(): I80F48 {
+    return I80F48.fromI64(this.netBorrowLimitPerWindowQuote).div(this.price);
+  }
+
+  getNetBorrowLimitPerWindowUi(): number {
+    return toUiDecimals(
+      I80F48.fromI64(this.netBorrowLimitPerWindowQuote).div(this.price),
+      this.mintDecimals,
+    );
+  }
+
+  getMaxWithdrawNative(vaultBalance: BN, userDeposits = ZERO_I80F48()): I80F48 {
+    // borrow is capped to minVaultToDepositsRatio and netBorrowLimitPerWindowQuote
+    const maxWithdrawFromVault = this.nativeDeposits().mul(
+      I80F48.fromNumber(1 - this.minVaultToDepositsRatio),
+    );
+
+    // userDeposits can exceed maxWithdrawFromVault
+    let maxBorrow = maxWithdrawFromVault.sub(userDeposits).min(ZERO_I80F48());
+    maxBorrow = maxBorrow.min(this.getNetBorrowLimitPerWindow());
+    maxBorrow = maxBorrow.div(ONE_I80F48().add(this.loanOriginationFeeRate));
+
+    // user deposits can always be withdrawn
+    // vaults can be depleted
+    return maxBorrow.add(userDeposits).min(I80F48.fromI64(vaultBalance));
+  }
+
+  getWhenNextBorrowLimitWindowStartsTs(): number {
```

Comment:
It looks like this function returns the number of seconds until the next window starts. From the name I'd expect an absolute timestamp.

Reply:
renamed to getTimeToNextBorrowLimitWindowStartsTs, wdyt?